go/pserver/service.go

```diff
@@ -51,6 +56,88 @@ type Service struct {
 	optMap map[string]*optimizer
 }
 
+// CheckpointMeta saves the checkpoint information
+type CheckpointMeta struct {
+	UUID      string `json:"uuid"`
+	MD5       string `json:"md5"`
+	Timestamp string `json:"timestamp"`
+}
+
+// ParameterCheckpoint saves parameter checkpoint
+type ParameterCheckpoint struct {
+	ParameterWithConfig
```

Comment:
According with: https://golang.org/ref/spec#Struct_types, Field without a explicit field name is an anonymous field, so I can initialize the `Param` filed of `ParameterWithConfig` as `ParameterCheckpoint.Param = xxx`